packages/yew/src/suspense/component.rs

```diff
@@ -12,95 +7,139 @@ pub struct SuspenseProps {
-            Self::Message::Resume(ref m) => {
-                let suspensions_len = self.suspensions.len();
-                self.suspensions.retain(|n| m != n);
+        }
 
-                suspensions_len != self.suspensions.len()
+        fn view(&self, ctx: &Context<Self>) -> Html {
+            let BaseSuspenseProps { children, fallback } = (*ctx.props()).clone();
+            let children = html! {<>{children}</>};
+
+            match fallback {
+                Some(fallback) => {
+                    let vsuspense = VSuspense::new(
+                        children,
+                        fallback,
+                        !self.suspensions.is_empty(),
+                        // We don't need to key this as the key will be applied to the component.
+                        None,
+                    );
+
+                    VNode::from(vsuspense)
+                }
+                None => children,
             }
         }
     }
 
-    fn view(&self, ctx: &Context<Self>) -> Html {
-        let SuspenseProps {
-            children,
-            fallback: fallback_vnode,
-            key,
-        } = (*ctx.props()).clone();
-
-        let children_vnode =
-            VNode::from(VList::with_children(children.into_iter().collect(), None));
-
-        let vsuspense = VSuspense::new(
-            children_vnode,
-            fallback_vnode,
-            self.detached_parent.clone(),
-            !self.suspensions.is_empty(),
-            key,
-        );
-
-        VNode::from(vsuspense)
+    impl BaseSuspense {
+        pub(crate) fn suspend(&self, s: Suspension) {
+            self.link.send_message(BaseSuspenseMsg::Suspend(s));
+        }
+
+        pub(crate) fn resume(&self, s: Suspension) {
+            self.link.send_message(BaseSuspenseMsg::Resume(s));
+        }
+    }
+
+    /// Suspend rendering and show a fallback UI until the underlying task completes.
+    #[function_component]
+    pub fn Suspense(props: &SuspenseProps) -> Html {
+        let SuspenseProps { children, fallback } = props.clone();
+
+        let fallback = html! {
+            <BaseSuspense fallback={None}>
+                {fallback}
+            </BaseSuspense>
+        };
+
+        html! {
+            <BaseSuspense {fallback}>
+                {children}
+            </BaseSuspense>
+        }
     }
 }
 
 #[cfg(any(feature = "csr", feature = "ssr"))]
-impl Suspense {
-    pub(crate) fn suspend(&self, s: Suspension) {
-        self.link.send_message(SuspenseMsg::Suspend(s));
-    }
+pub use feat_csr_ssr::*;
 
-    pub(crate) fn resume(&self, s: Suspension) {
-        self.link.send_message(SuspenseMsg::Resume(s));
+#[cfg(not(any(feature = "ssr", feature = "csr")))]
+mod feat_no_csr_ssr {
+    use super::*;
+
+    use crate::function_component;
+
+    /// Suspend rendering and show a fallback UI until the underlying task completes.
+    #[function_component]
+    pub fn Suspense(_props: &SuspenseProps) -> Html {
+        Html::default()
```

Comment:
This is here to be replaced when SSR hydration is a thing, right?

Reply:
This is a skeleton code for libraries when they don't have any renderer enabled (which means that they never gets rendered).

If we apply feature flags on the actual Suspense component multiple feature flags will be needed.

Reply:
I think in the long term, it may be beneficial to separate rendering logic into a separate crate so we don't keep adding feature flags.